python/core/auto_generated/qgsdataitem.sip.in

```diff
@@ -404,6 +404,17 @@ Sets a custom sorting ``key`` for the item.
     void moveToThread( QThread *targetThread );
 %Docstring
 Move object and all its descendants to thread
+%End
+
+    virtual QgsAbstractDatabaseProviderConnection *databaseConnection() const /Factory/;
```

Comment:
Would we be better off returning QgsAbstractProviderConnection here, so that in future we can potentially also use this for other connection types?

Reply:
Yes, it makes sense. The drawback is that It would require additional type checking logic in the client code though: the existing client code is specialized to handle DBs (create table etc.).

I'm undecided if it would be better to leave this method and just add a generic one `QgsAbstractProviderConnection *connection()`.

In python you would need to duck type it which is not a good programming pattern.

Reply:
Anyway, this started as a bugfix and ended up to a new API, can we delay this further refactoring to a later release? 

I think we'd better keep `databaseConnection` for convience anyway, the most part of the connections API is being used for DB operations from the browser and from DB manager.

Side note: the connections API idea actually started as an API to load/store connection information from the QgsSettings to avoid each provider to reinvent the whell but I see that nobody is actually using it for what it was initially planned.